keras_nlp/utils/text_generation_test.py

```diff
@@ -970,10 +1005,15 @@ def call(self, inputs):
                 )
                 return generated
 
-        inputs = tf.constant([[0, 1], [1, 2]])
+        if ragged:
```

Comment:
these paramers are getting a little hard to read. can't just add an extra assert and avoid the extra parameter?

```
self.assertAllEqual(model.predict(ragged_inputs), ragged_outputs)
self.assertAllEqual(model.predict(dense_inputs), dense_inputs)
```

also, unrelated to this PR, but the dense outputs is hard to read. maybe we can turn this

```
expected_outputs = tf.tile([[3], [0]], [1, max_length - 2])
expected_outputs = tf.concat([inputs, expected_outputs], axis=1)
```
into something like
```
expected_outputs = tf.constant([[0, 1, 3, 3, 3], [1, 2, 0, 0, 0]])
```

please make these changes to all the test in this files where you are adding a ragged parameterized case.

Reply:
Ohh, I added it as a parameter because the UT for ragged input with `jit_compile = True` fails.

Reply:
Ah right! That's annoying we need to so many workarounds, but agreed this is the best option!